textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/source/ScanBasedSourceOperator.java

```diff
@@ -0,0 +1,113 @@
+package edu.uci.ics.textdb.dataflow.source;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.analysis.standard.StandardAnalyzer;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.ScoreDoc;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.FSDirectory;
+
+import edu.uci.ics.textdb.api.common.IField;
+import edu.uci.ics.textdb.api.common.ITuple;
+import edu.uci.ics.textdb.api.dataflow.ISourceOperator;
+import edu.uci.ics.textdb.common.StringField;
+import edu.uci.ics.textdb.common.exception.DataFlowException;
+import edu.uci.ics.textdb.common.exception.ErrorMessages;
+import edu.uci.ics.textdb.dataflow.common.DataTuple;
+import edu.uci.ics.textdb.dataflow.constants.LuceneConstants;
+
+/**
+ * Created by chenli on 3/28/16.
+ */
+public class ScanBasedSourceOperator implements ISourceOperator {
+    
+    private String dataDir;
+    private List<String> schema;
+    private int cursor = -1;
+    private IndexSearcher indexSearcher;
+    private ScoreDoc[] scoreDocs;
+    private IndexReader indexReader;
+    
+    public ScanBasedSourceOperator(String dataDirectory, List<String> schema) {
+        this.dataDir = dataDirectory;
+        this.schema = schema;
+    }
+
+    @Override
+    public void open() throws DataFlowException {
+
+        try {
+            Directory directory = FSDirectory.open(Paths.get(dataDir));
+            indexReader = DirectoryReader.open(directory);
+            indexSearcher = new IndexSearcher(indexReader);
+            //TODO we don't need analyzer for scanning all the docs!!
+            Analyzer analyzer = new StandardAnalyzer();

```

Comment:
Looks like there is no other way of constructing the QueryParser object without passing an Analyzer. Will remove the TODO